benchmark/camelyon/benchmark_metrics.py

```diff
@@ -0,0 +1,132 @@
+        if not self == other:
+            raise PerformanceError(f"Performances are not equals with 1e-{N_DIGITS_TOL} tol \n{self}\n{other}")
+
+    @staticmethod
+    def _cast_performances(performances: RawPerformances) -> Dict[str, float]:
+        """
+        Cast raw performances.
+
+        Args:
+            performances (RawPerformances): Raw experiment performances
+
+        Returns:
+            (Dict[str, float]): Performances
+        """
+        if not performances:
+            raise TypeError(f"Found {performances}, expected {RawPerformances}")
+
+        perf_list = list(performances.values())
+        if not all([p == perf_list[0] for p in perf_list[1:]]):
+            raise PerformanceError("Performances from all clients are expected to be equal.")
+
+        return {key: round(val, N_DIGITS_TOL) for key, val in perf_list[0].items()}
+
+
+def assert_expected_results(cl_metrics: BenchmarkResults, sa_metrics: BenchmarkResults, mode: str) -> None:
+    """
+    Assert benchmark results are the one expected, ie:
+        - Assert substrafl and pure-torch performances are equals
+        - Assert performances values are the one expected regarding the execution mode
+
+    Args:
+        cl_metrics (BenchmarkResults): Benchmark results for the substrafl experiment
+        sa_metrics (BenchmarkResults): Benchmark results for the pure torch experiment
+        mode:
+
+    Returns:
+        None
+
+    Raises:
+        PerformanceError if results are not the one expected.
+    """
+    cl_metrics.assert_performance_equals(other=sa_metrics)
+
+    if mode == "subprocess":
+        expected_performances = {"Accuracy": 0.5, "ROC AUC": 1.0}
+    elif mode == "remote":
+        expected_performances = {"Accuracy": 1.0, "ROC AUC": 1.0}
+    else:
+        return
+
+    cl_metrics.assert_performance_equals(
+        other=BenchmarkResults(name=f"expected_{mode}", exec_time=0, performances={"0": expected_performances})
+    )
+
+    return
+
+
+def get_metrics_from_substra_performances(performances: Performances) -> BenchmarkResults:
+    """
+    Get benchmark performances.
+
+    Args:
+        performances (Performances): substra Performances
+
+    Returns:
+        typing.Dict[str, typing.Dict[str, float]
+
+    Raises:
+        PerformanceError:
+            - if the number of worker in Performances differ from the number of Substra clients
+            - if one identifier is evaluated several times for one worker
+    """
+    execution_time: datetime.timedelta = performances.compute_plan_end_date[0] - performances.compute_plan_start_date[0]
+    execution_time: float = execution_time.total_seconds()
+
+    workers, metrics, values = performances.worker, performances.identifier, performances.performance
+    performances = {}
+
+    for org, metric, value in list(zip(workers, metrics, values)):
+        if org not in performances:
+            performances[org] = {}
```

Comment:
Wondering if it would be more readable to use `defaultdict(dict)` for `performances` instead 🤔 
(But I might tend to overuse defaultdict, so no strong opinion here)